### PR TITLE
Enable sorting on the `/evaluations` endpoint

### PR DESCRIPTION
**Pull request by ntlind**
## Improvements
- Allow users to pass-in the names of metrics that they want to sort on using a new parameter, `metrics_to_sort_by`, when calling `get_evaluations()`. Users can specify multiple metrics per task type if they wish.
- Evaluations are always sorted first by task type, then by `metrics_to_sort_by` if not None, then by `created_at` (newest evaluations near the top). Previously, evaluations were only sorted by `created_at`.

Examples:
```python
# no sorting
unsorted_evaluations = client.get_evaluations(
        datasets=["clf_dataset", "seg_dataset", "det_dataset"]
)

...

# sort by accuracy (which is at the label_key level) and precision (which is at the label level)
evaluations = client.get_evaluations(
        datasets=["clf_dataset", "seg_dataset", "det_dataset"],
        metrics_to_sort_by={
            "Accuracy": "class",
            "Precision": {"key": "class", "value": "1"},
        },
)

...

# sort all metrics across all task types
evaluations = client.get_evaluations(
        datasets=["clf_dataset", "seg_dataset", "det_dataset"],
        metrics_to_sort_by={
            "Accuracy": "class",
            "IOU": {"key": "k2", "value": "v2"},
            "mAPAveragedOverIOUs": "class",
        },
)

...

```

### CODE REVIEW

File: ts-client/src/ValorClient.ts [hunk 1]
```diff
@@ -6,9 +6,21 @@ import axios, { AxiosInstance } from 'axios';
  * @param value The value to type check.
  * @returns A boolean result.
  */
-function isGeoJSONObject(value: any): value is { type: string, coordinates: any } {
-  const geoJSONTypes: string[] = ["point", "linestring", "polygon", "multipoint", "multilinestring", "multipolygon"];
-  return typeof value === 'object' && value !== null && 'type' in value && geoJSONTypes.includes((value.type as string).toLowerCase());
+function isGeoJSONObject(value: any): value is { type: string; coordinates: any } {
```
Comment (ntlind) [inline ts-client/src/ValorClient.ts:9]: many of these changes were caused by running Prettier on this doc. I'll call-out actual functional changes via comments for convenience

File: ts-client/src/ValorClient.ts [hunk 2]
```diff
@@ -468,12 +490,17 @@ export class ValorClient {
    * Bulk fetches evaluation by array of ids
    *
    * @param id id of the evaluation
+   * @param metricsToSortBy An array of metrics to sort the evaluations by. Should only include dataset-level metrics (e.g., mAccuracy).
```
Comment (ntlind) [inline ts-client/src/ValorClient.ts:493]: changed this function

File: ts-client/src/ValorClient.ts [hunk 3]
```diff
@@ -482,25 +509,37 @@ export class ValorClient {
    * Fetches all evaluations associated to given models
    *
    * @param modelNames names of the models
+   * @param metricsToSortBy An array of metrics to sort the evaluations by. Should only include dataset-level metrics (e.g., mAccuracy).
```
Comment (ntlind) [inline ts-client/src/ValorClient.ts:512]: added metricsToSortBy to this function

File: ts-client/src/ValorClient.ts [hunk 4]
```diff
@@ -482,25 +509,37 @@ export class ValorClient {
...
+      models: modelNames.join(','),
+      metrics_to_sort_by: metricsToSortBy != null ? metricsToSortBy.join(',') : null
+    });
   }
 
   /**
    * Fetches all evaluations associated to given datasets
    *
    * @param datasetNames names of the datasets
+   * @param metricsToSortBy An array of metrics to sort the evaluations by. Should only include dataset-level metrics (e.g., mAccuracy).
```
Comment (ntlind) [inline ts-client/src/ValorClient.ts:531]: added metricsToSortBy to this function

File: ts-client/tests/ValorClient.test.ts [hunk 5]
```diff
@@ -227,14 +246,14 @@ test('evaluation methods', async () => {
   );
   // check we can get evaluations by model names
   expect((await client.getEvaluationsByModelNames([modelNames[0]])).length).toBe(2);
-  expect((await client.getEvaluationsByModelNames(modelNames)).length).toBe(4);
+  expect(
+    (await client.getEvaluationsByModelNames(modelNames, ['mAccuracy'])).length
```
Comment (ntlind) [inline ts-client/tests/ValorClient.test.ts:250]: added a quick test to make sure metricsToSortBy runs as expected. didn't add detailed tests here as a) all of these evaluations have the same mAccuracy and b) I'd already written detailed integration tests in the python client

File: ts-client/src/ValorClient.ts (hunk 1 above)
Comment (ekorman) [reply ts-client/src/ValorClient.ts:9]: good with me!

**Review by ekorman — APPROVED** [review]

**Pull request merged**